cmd/deploy_test.go

```diff
@@ -270,3 +274,88 @@ runtime: go`,
 		})
 	}
 }
+
+func Test_namespaceCheck(t *testing.T) {
+	tests := []struct {
+		name      string
+		registry  string
+		namespace string
+		funcFile  string
+		expectNS  string
+	}{
+		{
+			name:     "first deployment(no ns in func.yaml), not given via cli, expect write in func.yaml",
+			registry: "docker.io/4141gauron3268",
+			expectNS: "test-ns-deploy",
+			funcFile: `name: test-func
+runtime: go`,
+		},
+		{
+			name:     "ns in func.yaml, not given via cli, current ns matches func.yaml",
+			registry: "docker.io/4141gauron3268",
+			expectNS: "test-ns-deploy",
+			funcFile: `name: test-func
+namespace: "test-ns-deploy"
+runtime: go`,
+		},
+		{
+			name:      "ns in func.yaml, given via cli (always override)",
+			namespace: "test-ns-deploy",
+			expectNS:  "test-ns-deploy",
+			registry:  "docker.io/4141gauron3268",
+			funcFile: `name: test-func
+namespace: "non-default"
+runtime: go`,
+		},
+		{
+			name:     "ns in func.yaml, not given via cli, current ns does NOT match func.yaml",
+			registry: "docker.io/4141gauron3268",
+			expectNS: "non-default",
+			funcFile: `name: test-func
+namespace: "non-default"
+runtime: go`,
+		},
+	}
+
+	// create mock kubeconfig with set namespace as 'default'
+	defer WithEnvVar(t, "KUBECONFIG", fmt.Sprintf("%s/testdata/kubeconfig_deploy_namespace", cwd()))()
+
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+			deployer := mock.NewDeployer()
+			defer Fromtemp(t)()
+			cmd := NewDeployCmd(NewClientFactory(func() *fn.Client {
+				return fn.New(
+					fn.WithDeployer(deployer))
+			}))
+
+			// set namespace argument if given & reset after
+			cmd.SetArgs([]string{}) // Do not use test command args
+			viper.SetDefault("namespace", tt.namespace)
+			viper.SetDefault("registry", tt.registry)
+			defer viper.Reset()
+
+			// set test case's func.yaml
+			if err := os.WriteFile("func.yaml", []byte(tt.funcFile), os.ModePerm); err != nil {
+				t.Fatal(err)
+			}
+
+			ctx := context.TODO()
```

Comment:
this could be `context.Background()`

Reply:
Yeah, use context.Background() everywhere in the test.